client/components/dot-pager/index.jsx

```diff
@@ -74,13 +74,18 @@ export const DotPager = ( {
 	hasDynamicHeight = false,
 	children,
 	className,
+	onSetCurrentPage,
 	...props
 } ) => {
 	const [ currentPage, setCurrentPage ] = useState( 0 );
 	const [ pagesStyle, setPagesStyle ] = useState();
 	const pagesRef = useRef();
 	const [ resizeObserver, sizes ] = useResizeObserver();
 
+	useEffect( () => {
+		onSetCurrentPage && onSetCurrentPage( currentPage );
+	}, [ currentPage, onSetCurrentPage ] );
```

Comment:
Sort of related to the above comment about the prop name, rather than spying on local state like this, it feels more straight forward to notify the parent when the change event happens and then carry on. Like this.

```typescript
<Controls
  // ...
  setCurrentPage={ ( index ) => {
    setCurrentpage( index );
    onPageSelected && onPageSelected( index );
  } }
>
```

It does put the onus on the parent for dealing with the the initial page render though i.e. it wouldn't call `onPageSelected( 0 )` on initial render.

Reply:
I think `onPageSelected` or `onPageShow` is great! However, I prefer to keep the current way to handle page show events, but I'm okay with both. On the other hand, if the user switches the page from first to last and the user closes the page before the transition is finished, do we need to send a track event for this case?

Reply:
> On the other hand, if the user switches the page from first to last and the user closes the page before the transition is finished, do we need to send a track event for this case?

No I don't think this would matters. The idea of an "impression" is pretty fuzzy. We don't really know whether the user ever really _looks_ at a card or not (e.g. we send an event for the mobile card but it's down the bottom so they might not interact with it). So I think however the tracks events happen in this case would be fine.

Reply:
Okay! I just think it's a little bit inconsistent because here we change to send tracks events when the user goes to that card but somewhere as you mentioned that we send the tracks events we don't really know whether the user ever really looks at a card or not.

Reply:
😅 Fair point. I think my main worry with how we were sending impression events for every card when the page loaded was that it made the tracks events almost useless. There'd be no point analysing them because every card would have close to the same number of events. This PR makes them a little bit better than that.
But it is a bit inconsistent.